### ISSUE TITLE
Use hi-res OS/2 timer in QElapsedTime and QTimerInfoList

### ISSUE THREAD

**Issue by dmik** [bitwiseworks/qtbase-os2#53]
Currently, we use the generic `QElapsedTime` implementation which internally uses `QDateTime::currentMSecsSinceEpoch()` which in turn uses LIBC `gettimeofday` which internally uses an OS/2 system timer (QSV_MS_COUNT, a 32-bit free running counter representing the number of milliseconds since the reboot). It has a granularity of around 18 ms which is far from being accurate according to the current standards (smooth animation and so on).

On OS/2, there is a high-resolution 64-bit timer provided by `DosTmrQueryTime` whose resolution is around 1 ms. We should use it instead of the 32-bit ms counter in both `QElapsedTime` and in `QTimerInfoList` which is the code from the Unix platform that we use to implement `QTimer` and friends on OS/2.

**Comment by dmik** [comment]
Note that good test cases for this change are `tst_QTimer` and `tst_QSocketNotifier` corelib auto-tests. They should be re-evaluated once it's made.

**Comment by dmik** [comment]
Another test case to re-evaluate is `tst_QPauseAnimation` where some test failures are ignored on OS/2 due to BAD_TIMER_RESOLUTION (see the above commit where OS/2 was added to the ignored list). Using the high-res timer may make them work.

**Comment by dmik** [comment]
One should check comments in #24 (which is a duplicate) when working on this one.

**Comment by SilvanScherrer** [comment]
I guess this should all work now, as acpi is fixed.

**Comment by dmik** [comment]
Yes, but first one needs to use the high-res timer in Qt :) BTW, as LIBCn [supports](https://github.com/bitwiseworks/libc/issues/53) `clock_gettime(CLOCK_MONOTONIC)` now which interfaces with this high-res timer, we should use that instead of `DosTmrQueryTime` directly.

**Comment by dmik** [comment]
Basically we just need to enable the Unix impleemntation (`qelapsedtimer_unix.cpp`) on OS/2 that uses `clock_gettime(CLOCK_MONOTONIC)`.

**Comment by SilvanScherrer** [comment]
Sound like a fast but good fix 😅
